q2_keegan/vanilla/tests/test_format.py

```diff
@@ -0,0 +1,37 @@
+import os
+import os.path
+import unittest
+
+from q2_keegan.vanilla import VanillaBeanFmt, VanillaBeanDirFmt
+from qiime2.plugin.testing import TestPluginBase
+from qiime2.plugin import ValidationError
+
+class TestVanillaFormats(TestPluginBase):
+    package = 'q2_keegan.vanilla.tests'
+
+    def test_vanilla_format_validate_positive(self):
+        filenames = ['single-row-valid.txt', 'multi-row-valid.txt',
+                     'multi-row-w-blank-lines-valid.txt',
+                     ]
+        filepaths = [self.get_data_path(os.path.join('vanilla', filename))
+                        for filename in filenames]
+
+        for filpath in filepaths:
+            format = VanillaBeanFmt(filpath, mode='r')
+
+            format.validate()
```

Comment:
Stylistically, no need to iterate over these elements twice, you can combine all of this into one for loop:

```suggestion
        for filename in filenames:
            filepath = self.get_data_path(filename)
            format = VanillaBeanFmt(filepath, mode='r')
            format.validate()
```